recipes/aws-c-auth/all/conanfile.py

```diff
@@ -44,19 +47,22 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("aws-c-common/0.8.2", transitive_headers=True, transitive_libs=True)
-        self.requires("aws-c-cal/0.5.13")
-        if Version(self.version) < "0.6.17":
-            self.requires("aws-c-io/0.10.20", transitive_headers=True)
+        if Version(self.version) <= "0.6.17":
+            self.requires("aws-c-common/0.8.2", transitive_headers=True, transitive_libs=True)
+            self.requires("aws-c-cal/0.5.13")
+            self.requires("aws-c-io/0.10.20", transitive_headers=True, transitive_libs=True)
             self.requires("aws-c-http/0.6.13", transitive_headers=True)
```

Comment:
This changes restored the previously used versions for `0.6.17`. Locally the tests now pass.

```suggestion
            if Version(self.version) < "0.6.17":
                self.requires("aws-c-io/0.10.20", transitive_headers=True, transitive_libs=True)
                self.requires("aws-c-http/0.6.13", transitive_headers=True)
            else:
                self.requires("aws-c-io/0.13.4", transitive_headers=True, transitive_libs=True)
                self.requires("aws-c-http/0.6.22", transitive_headers=True)
```